src/hooks/useEditStake.ts

```diff
@@ -0,0 +1,133 @@
+                    )?.joinDate as string,
+                    'X',
+                )
+                const minLeaveDate = joinDate.add(
+                    sponsorship.minimumStakingPeriodSeconds,
+                    'seconds',
+                )
+
+                await editStakeModal.pop({
+                    currentStake: currentStake.amount,
+                    operatorId: myOperatorQuery.data.id,
+                    operatorBalance: myOperatorQuery.data.freeFundsWei.toString(),
+                    tokenSymbol: tokenInfo.symbol,
+                    decimals: tokenInfo.decimals,
+                    leavePenalty: leavePenaltyWei.toString(),
+                    minLeaveDate: minLeaveDate.format('YYYY-MM-DD HH:mm'),
+                    hasUndelegationQueue: myOperatorQuery.data.queueEntries.length > 0,
+                    onSubmit: async (
+                        amount: string,
+                        difference: string,
+                        forceUnstake = false,
+                    ) => {
+                        const differenceBN = toBN(difference)
+                        if (differenceBN.isGreaterThanOrEqualTo(0)) {
+                            await stakeOnSponsorship(
+                                sponsorship.id,
+                                difference,
+                                myOperatorQuery.data?.id as string,
+                                'Increase stake on sponsorship',
+                            )
+                        } else if (
+                            forceUnstake &&
+                            (await confirmationModal({
+                                title: 'Your stake will be slashed',
+                                description: `Your minimum staking period is still ongoing and ends on ${minLeaveDate.format(
+                                    'YYYY-MM-DD HH:mm',
+                                )}. If you unstake now, you will lose ${leavePenalty} ${
+                                    tokenInfo.symbol
+                                }`,
+                                proceedLabel: 'Proceed anyway',
+                                cancelLabel: 'Cancel',
+                            }))
+                        ) {
+                            await forceUnstakeFromSponsorship(
+                                sponsorship.id,
+                                myOperatorQuery.data?.id as string,
+                            )
+                        } else if (!forceUnstake) {
+                            await reduceStakeOnSponsorship(
+                                sponsorship.id,
+                                amount,
+                                myOperatorQuery.data?.id as string,
+                                amount === '0'
+                                    ? 'Unstake from sponsorship'
+                                    : 'Reduce stake on sponsorship',
+                            )
+                        }
```

Comment:
This whole thing.. Let's short-circuit it.

```ts
if (differenceBN.isGreaterThanOrEqualTo(0)) {
    return stakeOnSponsorship(/* … */)
}

if (!forceUnstake) {
    return reduceStakeOnSponsorship(/* … */)
}

if (!confirm(/* modal props */)) {
    return
}

return forceUnstakeFromSponsorship(/* … */)
```

And you also can change `async onSubmit` to just `onSubmit` cause of no `await`.

Reply:
it still needs to be async as we'd need to await for `confirm` function